cpp/memilio/CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+configure_file(config_internal.h.in memilio/config_internal.h)
+
+add_library(memilio
+    config.h
+    epidemiology/populations.h
+    epidemiology/damping.cpp
+    epidemiology/damping.h
+    epidemiology/uncertain_matrix.h    
+    epidemiology/uncertain_matrix.cpp
+    epidemiology/contact_matrix.h    
+    epidemiology/contact_matrix.cpp
+    epidemiology/damping_sampling.h    
+    epidemiology/damping_sampling.cpp
+    epidemiology/dynamic_npis.h    
+    epidemiology/dynamic_npis.cpp
+    epidemiology/regions.h
+    epidemiology/regions.cpp
+    epidemiology/holiday_data_de.ipp
+    compartments/compartmentalmodel.h
+    compartments/simulation.h
+    io/io.h
+    io/io.cpp
+    io/hdf5_cpp.h
+    io/json_serializer.h
+    io/json_serializer.cpp
+    io/mobility_io.h
+    io/mobility_io.cpp
+    math/euler.cpp
+    math/euler.h
+    math/smoother.h
+    math/adapt_rk.cpp
+    math/adapt_rk.h
+    math/integrator.h
+    math/integrator.cpp
+    math/eigen.h
+    math/eigen_util.h
+    math/matrix_shape.h
+    math/matrix_shape.cpp
+    mobility/mobility.h
+    mobility/mobility.cpp
+    mobility/graph_simulation.h
+    mobility/graph_simulation.cpp
+    mobility/graph.h
+    mobility/graph.cpp
+    utils/visitor.h
+    utils/uncertain_value.h
+    utils/uncertain_value.cpp
+    utils/index.h
+    utils/custom_index_array.h
+    utils/memory.h
+    utils/parameter_distributions.h
+    utils/time_series.h
+    utils/time_series.cpp
+    utils/span.h
+    utils/span.cpp
+    utils/type_safe.h
+    utils/type_safe.cpp
+    utils/parameter_set.h
+    utils/parameter_set.cpp
+    utils/date.h
+    utils/date.cpp
+    utils/random_number_generator.h
+    utils/random_number_generator.cpp
+)
```

Comment:
Just a comment (feel free to ignore). I find this list a bit intimidating. What do you think about either
1. using something like `file(GLOB_RECURSE source_list "*.cpp" "*.hpp")` , or
2. using `add_directory()` and amending the `source_list` by shorter lists in the `CMakeLists.txt` of the subdirectories io, utils, models, ...?

Reply:
> using something like file(GLOB_RECURSE source_list "*.cpp" "*.hpp")

this is not recommended (last I checked, maybe there is a way to make this work properly now?). The advantage of not having to change the CMakeLists is also the main drawback. Cmake doesn't detect new files automatically. So if you glob the sources like that, cmake configure must be run manually when a file is added or the new file is not built. This may be OK for the developer who adds the file, but it's weird for other developers. They will get unexpected linker errors if they pull changes from github because they don't know they need to run cmake. If the CMakeLists need to be modified every time a file is added, Cmake never needs to be run manually, it runs automatically when building. Since CI and local tests reliably detect when source files are missing from this list (the build just fails), the time saved for the person that adds files is not worth the additional work for everyone else.

> using add_directory() and amending the source_list by shorter lists in the CMakeLists.txt of the subdirectories io, utils, models, ..

I'm not opposed to that, although I wonder if this would be that much of an improvement? I think this means I'll always look in the wrong place, but maybe that's just me :smile: